### PR TITLE
Update claiming job logic to check credentials and availability

### PR DESCRIPTION
**Pull request by Sgitario**
Fix https://github.com/halkyonio/primaza-poc/issues/150

### CODE REVIEW

File: servicebox-app/src/main/java/io/halkyon/services/UpdateClaimJob.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+/**
+ * The claiming service will poll every new and pending claim to update the claim status according to the following logic:
+ *
+ * A claim can request a service in the form of `<service name>:<service version>`, for example: "mysql:3.6".
+ * If there is a service that matches the criteria of service name, plus service version, then the claim is linked to this service.
+ * Then, if this service has been found in a cluster (the service available flag is true), and there is at least one service
+ * credential, then the claim status will change to "Bind"
+ * this service will
+ * be linked to the claim and the claim status will change to "claimed". Otherwise, the status will be "pending".
+ * After a number of attempts have been made to find a suitable service, the claim status will change to "error".
```
Comment (aureamunoz) [inline servicebox-app/src/main/java/io/halkyon/services/UpdateClaimJob.java:27]: This needs to be updated as well

File: servicebox-app/src/main/java/io/halkyon/services/UpdateClaimJob.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    public static final String ERROR_MESSAGE_NO_CREDENTIALS_FOUND_IN_SERVICE_BOX = "Service '%s' has no credentials";
+    public static final String ERROR_MESSAGE_NO_SERVICE_FOUND_IN_CLUSTER = "Could not find a service with protocol '%s'";
+
+    @ConfigProperty(name = "servicebox.update-claim-job.max-attempts")
+    int maxAttempts;
+
+
+    /**
+     * This method will be executed at every `${servicebox.update-claim-job.poll-every}`.
+     * First, it will collect the list of all available services, and then will loop over the new or pending claims to link
```
Comment (aureamunoz) [inline servicebox-app/src/main/java/io/halkyon/services/UpdateClaimJob.java:42]: This method is not collecting all available services anymore....

File: servicebox-app/src/main/java/io/halkyon/services/UpdateClaimJob.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+ * Then, if this service has been found in a cluster (the service available flag is true), and there is at least one service
+ * credential, then the claim status will change to "Bind"
+ * this service will
+ * be linked to the claim and the claim status will change to "claimed". Otherwise, the status will be "pending".
+ * After a number of attempts have been made to find a suitable service, the claim status will change to "error".
+ */
+@ApplicationScoped
+public class UpdateClaimJob {
+
+    public static final String ERROR_MESSAGE_NO_SERVICE_FOUND_IN_SERVICE_BOX = "Service '%s' not registered";
```
Comment (aureamunoz) [inline servicebox-app/src/main/java/io/halkyon/services/UpdateClaimJob.java:32]: I would avoid `SERVICE_BOX` in the name.... 
```suggestion
    public static final String ERROR_MESSAGE_NO_SERVICE_FOUND_IN_CATALOGUE = "Service '%s' not registered";
```

File: servicebox-app/src/main/java/io/halkyon/services/UpdateClaimJob.java [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    public static final String ERROR_MESSAGE_NO_CREDENTIALS_FOUND_IN_SERVICE_BOX = "Service '%s' has no credentials";
```
Comment (aureamunoz) [inline servicebox-app/src/main/java/io/halkyon/services/UpdateClaimJob.java:33]:
```suggestion
    public static final String ERROR_MESSAGE_NO_CREDENTIALS_FOUND = "Service '%s' has no credentials";
```

**Review by aureamunoz — CHANGES_REQUESTED** [review]

File: servicebox-app/src/main/java/io/halkyon/services/UpdateClaimJob.java (hunk 3 above)
Comment (Sgitario) [reply servicebox-app/src/main/java/io/halkyon/services/UpdateClaimJob.java:32]: Changed to ERROR_MESSAGE_NO_SERVICE_REGISTERED.

File: servicebox-app/src/main/java/io/halkyon/services/UpdateClaimJob.java (hunk 2 above)
Comment (Sgitario) [reply servicebox-app/src/main/java/io/halkyon/services/UpdateClaimJob.java:42]: well spotted! updated.

File: servicebox-app/src/main/java/io/halkyon/services/UpdateClaimJob.java (hunk 1 above)
Comment (Sgitario) [reply servicebox-app/src/main/java/io/halkyon/services/UpdateClaimJob.java:27]: there were some left overs here, my bad. it should be fixed now.

File: servicebox-app/src/main/java/io/halkyon/services/UpdateClaimJob.java [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+    @ConfigProperty(name = "servicebox.update-claim-job.max-attempts")
+    int maxAttempts;
+
+
+    /**
+     * This method will be executed at every `${servicebox.update-claim-job.poll-every}`.
+     * First, it will collect the list of all services, and then will loop over the new or pending claims to link
+     * the service if the criteria matches.
+     */
```
Comment (aureamunoz) [inline servicebox-app/src/main/java/io/halkyon/services/UpdateClaimJob.java:45]: It's still out of date, no?

**Review by aureamunoz — CHANGES_REQUESTED** [review]

**Comment by Sgitario** [comment]
I need to merge this to continue with other issues.
Let's address the logic if needed in separate pull requests.

File: servicebox-app/src/main/java/io/halkyon/services/UpdateClaimJob.java (hunk 5 above)
Comment (aureamunoz) [reply servicebox-app/src/main/java/io/halkyon/services/UpdateClaimJob.java:45]: The method is not collecting the list of all services anymore....

**Pull request merged**